src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -0,0 +1,190 @@
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import com.google.sps.data.UserInfo;
+
+/** Servlet that returns events sorted by most recent timestamp */
+@WebServlet("/user")
+public class UserServlet extends HttpServlet {
+
+  static final String CHALLENGE_ID_PARAM = "chal";
+  static final String CHALLENGE_STATUS_PARAM = "stat";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    
+    String userId = userService.getCurrentUser().getUserId();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Query query = new Query(UserInfo.DATA_TYPE).setFilter(new FilterPredicate(UserInfo.ID, FilterOperator.EQUAL, userId));
+    Entity entity = datastore.prepare(query).asSingleEntity();
+
+    if (entity == null) {
+        doPost(request, response);
+        entity = datastore.prepare(query).asSingleEntity();
+    }
+
+    UserInfo user = convertEntitytoUserInfo(entity, userId);
+
+    response.setContentType("application/json; charset=UTF-8");
+    response.setCharacterEncoding("UTF-8");
+
+    String json = convertToJson(user);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      UserService userService = UserServiceFactory.getUserService();
+      String userId = userService.getCurrentUser().getUserId();
+      String userNickname = request.getParameter(UserInfo.NICKNAME);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); 
+
+      Long currentChallengeId = 0L;
+      ArrayList<Integer> challengeStatuses = new ArrayList<Integer>(Collections.nCopies(3, 0));
+
+      Entity userEntity = new Entity(UserInfo.DATA_TYPE);
+      userEntity.setProperty(UserInfo.ID, userId);
+      userEntity.setProperty(UserInfo.NICKNAME, userNickname);
+      userEntity.setProperty(UserInfo.CREATED_EVENTS, Collections.emptyList());
+      userEntity.setProperty(UserInfo.BOOKMARKED_EVENTS, Collections.emptyList());
+      userEntity.setProperty(UserInfo.CURRENT_CHALLENGE, currentChallengeId);
+      userEntity.setProperty(UserInfo.CHALLENGE_STATUSES, challengeStatuses);
+
+      datastore.put(userEntity);
+
+      response.sendRedirect("/feed.html");
+  }
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      UserService userService = UserServiceFactory.getUserService();
+      String userId = userService.getCurrentUser().getUserId();
+
+      String challengeIdParam = request.getParameter(CHALLENGE_ID_PARAM);
+      String statusParam = request.getParameter(CHALLENGE_STATUS_PARAM);
+
+      Long challengeId = -1L;
```

Comment:
Any reason challengeId needs to be assigned as -1?

Reply:
Used it for debugging and forgot to take it out